presto-main/src/main/java/com/facebook/presto/operator/unnest/UnnestOperatorBlockUtil.java

```diff
@@ -13,32 +13,304 @@
  */
 package com.facebook.presto.operator.unnest;
 
-import static java.lang.String.format;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.ArrayBlock;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.ByteArrayBlock;
+import com.facebook.presto.spi.block.ColumnarArray;
+import com.facebook.presto.spi.block.ColumnarMap;
+import com.facebook.presto.spi.block.ColumnarRow;
+import com.facebook.presto.spi.block.DictionaryBlock;
+import com.facebook.presto.spi.block.Int128ArrayBlock;
+import com.facebook.presto.spi.block.IntArrayBlock;
+import com.facebook.presto.spi.block.LongArrayBlock;
+import com.facebook.presto.spi.block.MapBlock;
+import com.facebook.presto.spi.block.RowBlock;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import com.facebook.presto.spi.block.ShortArrayBlock;
+import com.facebook.presto.spi.block.VariableWidthBlock;
+import com.facebook.presto.spi.function.OperatorType;
+import com.facebook.presto.spi.type.MapType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.spi.block.ArrayBlock.fromElementBlock;
+import static com.facebook.presto.spi.block.MapBlock.fromKeyValueBlock;
+import static com.facebook.presto.spi.block.MethodHandleUtil.compose;
+import static com.facebook.presto.spi.block.MethodHandleUtil.nativeValueGetter;
+import static com.facebook.presto.spi.block.RowBlock.fromFieldBlocks;
+import static com.facebook.presto.testing.TestingEnvironment.TYPE_MANAGER;
 
 final class UnnestOperatorBlockUtil
 {
     private UnnestOperatorBlockUtil() {}
 
-    private static final int DEFAULT_CAPACITY = 64;
-    // See java.util.ArrayList for an explanation
-    static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+    static Block insertNull(Block block, Type type)
```

Comment:
- What does this method do? My reading of the implementation is that it adds one null position at the end of the block. Perhaps, rename it to `appendNull`. 
- I'd extract this method into a separate commit. This is a self-contained unit-testable addition.